source/NVDAObjects/UIA/__init__.py

```diff
@@ -233,7 +236,9 @@ def _getControlFieldForObject(self, obj,isEmbedded=False,startOfNode=False,endOf
 		field["description"] = obj.description
 		field["level"] = obj.positionInfo.get("level")
 		if role == controlTypes.ROLE_TABLE:
-			field["table-id"] = 1 # FIXME
+			tableID=VARIANT()
+			obj.UIAElement._IUIAutomationElement__com_GetCurrentPropertyValue(UIAHandler.UIA_RuntimeIdPropertyId,byref(tableID))
+			field["table-id"] = obj

```

Comment:
The problem with using the NVDAObject is that it might get cached in speech for a long time if it happens to be the last table the user uses in a while. We discussed using RuntimeId instead and using a tree walker to get the table ancestor from a descendant object.

One problem is how to build an appropriate safearray. UIA expects an array of VT_I4, but by default, comtypes makes arrays of VT_VARIANT. A bit of reading suggests a Python "l" array object gets converted to the correct type; e.g. `array.array("l", (1, 2, 3))`. I'm not sure if comtypes wraps this in a VARIANT when passed as a parameter, but if not, you can just do this yourself:

```
>>> comtypes.automation.VARIANT(array.array("l", (1, 2, 3)))
VARIANT(vt=0x2003, (1, 2, 3))
```